### PR TITLE
Smile 1616 use cristind id to substitute contributor

### PR DESCRIPTION
**Pull request by anetteOlli**

### CODE REVIEW

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/model/cristin/Affiliation.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package no.sikt.nva.scopus.conversion.model.cristin;
```
Comment (anetteOlli) [inline create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/model/cristin/Affiliation.java:1]: Siden dette har blitt kopier fra nva-cristin-service, burde denne vært tatt inni  nva-datamodel slik at vi kan vi kan importere dette istedet for copy-pase.

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/CristinConnection.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    private Person logFailureAndReturnNull(Failure<Person> failure) {
+        logger.info(COULD_NOT_EXTRACT_CRISTIN_PERSON_ERROR_MESSAGE, failure.getException());
+        return null;
+    }
+
+    private Person getPiaAuthorResponse(String json) {
+        var gson = new Gson();
+        return gson.fromJson(json, Person.class);
```
[the remark this answers is not in the document]
Comment (anetteOlli) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/CristinConnection.java:57]: Are you referring to the "new" keyword that places object on the heap and does not support inversion of dependencies?
Are we using jackson incorrectly? ended up with var objectMapper = new ObjectMapper.

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/CristinConnection.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    private String getBodyFromResponse(HttpResponse<String> response) {
+        if (response.statusCode() != HttpURLConnection.HTTP_OK) {
+            logger.info(CRISTIN_RESPONDED_WITH_BAD_STATUS_CODE_ERROR_MESSAGE + response.statusCode());
+            throw new RuntimeException();
+        }
```
[the remark this answers is not in the document]
Comment (anetteOlli) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/CristinConnection.java:64]: I guess we can follow redirects, but at the same time, if we are ending up with somewhere else than where nva-cristin-service is running, I can't imagine we get something that can be converted to a Person.class.
Isn't the handler also an aws handler that will in practise never return "busy try again later"

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/model/cristin/Role.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        this.id = id;
+        this.labels = labels;
+    }
+
+    public URI getId() {
+        return id;
+    }
+
+    public Map<String, String> getLabels() {
+        return Objects.nonNull(labels) ? labels : Collections.emptyMap();
```
[the remark this answers is not in the document]
Comment (anetteOlli) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/model/cristin/Role.java:39]: This is also copy paste code from nva-cristin-service. But will fix.

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/ContributorExtractor.java [hunk 5]
```diff
@@ -162,13 +171,31 @@ private void generateContributorFromCollaborationTp(CollaborationTp collaboratio
...
-        identity.setId(piaConnection.getCristinID(authorTp.getAuid()));
+        identity.setId(cristinUri);
         return identity;
     }
 
+    private String determineContributorName(Person person) {
+        return person.getNames().stream().filter(this::isSurname).findFirst().map(
+            TypedValue::getValue).orElse(StringUtils.EMPTY_STRING) + NAME_DELIMITER
+               + person.getNames().stream().filter(this::isFirstName).findFirst().map(
+            TypedValue::getValue).orElse(StringUtils.EMPTY_STRING);
```
[the remark this answers is not in the document]
Comment (anetteOlli) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/ContributorExtractor.java]: Fikset

File: create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/CristinConnection.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (anetteOlli) [reply create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/CristinConnection.java]: Switched to jackson

**Review by brinxmat — APPROVED** [review]

**Review by gregorgabriel — APPROVED** [review]
no further comments. Bra jobber

**Pull request merged**